aws/resource_aws_kinesis_stream_consumer.go

```diff
@@ -0,0 +1,229 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsKinesisStreamConsumer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsKinesisStreamConsumerCreate,
+		Read:   resourceAwsKinesisStreamConsumerRead,
+		Delete: resourceAwsKinesisStreamConsumerDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				importParts, err := validateResourceAwsKinesisStreamConsumerImportString(d.Id())
+				if err != nil {
+					return nil, err
+				}
+				d.Set("name", importParts[0])
+				d.Set("stream_arn", importParts[1])
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"stream_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsKinesisStreamConsumerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kinesisconn
+	cn := d.Get("name").(string)
+	sa := d.Get("stream_arn").(string)
+
+	createOpts := &kinesis.RegisterStreamConsumerInput{
+		ConsumerName: aws.String(cn),
+		StreamARN:    aws.String(sa),
+	}
+
+	createOutput, err := conn.RegisterStreamConsumer(createOpts)
+	if err != nil {
+		return fmt.Errorf("Unable to create stream consumer: %s", err)
+	}
+
+	arn := aws.StringValue(createOutput.Consumer.ConsumerARN)
+
+	d.SetId(arn)
+	d.Set("arn", arn)
+
+	// No error, wait for ACTIVE state
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"CREATING"},
+		Target:     []string{"ACTIVE"},
+		Refresh:    streamConsumerStateRefreshFunc(conn, cn, sa),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for Kinesis Stream Consumer (%s) to become active: %s",
+			cn, err)
+	}
+
+	return resourceAwsKinesisStreamConsumerRead(d, meta)
+}
+
+func resourceAwsKinesisStreamConsumerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kinesisconn
+	cn := d.Get("name").(string)
+	sa := d.Get("stream_arn").(string)
+
+	state, err := readKinesisStreamConsumerState(conn, cn, sa)
```

Comment:
since we'll have the ARN stored as the ID, resource lookup will be updated to use that attribute and will be relocated in the service package as well